packages/mui-joy/src/Container/ContainerProps.ts

```diff
@@ -2,41 +2,60 @@ import * as React from 'react';
 import { OverrideProps } from '@mui/types';
 import { Breakpoint } from '@mui/system';
 import { SxProps } from '../styles/types';
+import { SlotProps, CreateSlotsAndSlotProps } from '../utils/types';
 
 export type ContainerSlot = 'root';
 
+export interface ContainerSlots {
+  /**
+   * The component that renders the root.
+   * @default 'div'
+   */
+  root: React.ElementType;
+}
+
+export type ContainerSlotsAndSlotProps = CreateSlotsAndSlotProps<
+  ContainerSlots,
+  {
+    root: SlotProps<'div', {}, ContainerOwnerState>;
+  }
+>;
+
```

Comment:
Let's leave system components out of this PR. You have to also update the implementation in `createContainer()`.